### PR TITLE
mrc-4763: Add endpoints to serve emulator data.

### PR DESCRIPTION
**Pull request by plietar**
If a `--emulator=PATH` argument is passed to the mintr main, two new endpoints are provided, one to serve a JSON configuration and the other to serve the ML model used by the emulator.

The endpoints are pretty agnostic to the actual format and contents of the data. This allows a lot of flexibility while we iterate over the client-side implementation of the emulator. In the future we may narrow this down, eg. by specifying a more restrictive JSON Schema for the configuration endpoint.

mintr can still function without any emulator data: if the `--emulator` argument is missing then the endpoints are not included, and calling them returns a 404.

### CODE REVIEW

**Review by EmmaLRussell — COMMENTED** [review]
Looks good to me, as far as I can tell! Couple of test failures in CI. 

I guess we're going to want some real emulator data and config rolled into the docker image which we can use to test the front end - but that will be in a separate branch?

File: R/api.R [hunk 1]
```diff
@@ -237,3 +241,29 @@ target_strategise <- function(db) {
...
+}
+
+target_emulator_config <- function(emulator_root) {
+  function() {
+    read_json(file.path(emulator_root, "config.json"))
+  }
+}
+
+endpoint_emulator_model <- function(emulator_root) {
+  print(emulator_root)
```
[the remark this answers is not in the document]
Comment (plietar) [reply R/api.R:258]: Oops yes. Removed now.

File: R/api.R [hunk 2]
```diff
@@ -237,3 +241,29 @@ target_strategise <- function(db) {
     )
   }
 }
+
+endpoint_emulator_config <- function(emulator_root) {
+  porcelain::porcelain_endpoint$new(
+    "GET", "/emulator/config", target_emulator_config(emulator_root),
+    returning = porcelain::porcelain_returning_json("Emulator"))
```
[the remark this answers is not in the document]
Comment (plietar) [reply R/api.R:248]: Done.

**Comment by plietar** [comment]
> I guess we're going to want some real emulator data and config rolled into the docker image which we can use to test the front end - but that will be in a separate branch?

In the short-term we probably won't have anything baked into the docker image. If you want to use the emulator you'll have to bind mount a folder into the container and pass it as an argument, something like `docker run -v $PWD/emulator:/emulator mrcide/mintr --emulator=/emulator`. I'll add a paragraph to the README about it.

In the long term we'll have some kind of pipeline that produces the emulator artefacts and we'll pull them in as part of the image build, similar to how the malariasimulation data is handled.

To be honest I hadn't considered the issue of question of writing tests for the frontend. My focus so far had just been on enabling us to iterate over the emulator models and its configuration. Maybe we want to start including an emulator model (even a bad one) sooner than I'd expected, just so we have something to run frontend tests against. I'll think more about that.

File: R/api.R [hunk 3]
```diff
@@ -237,3 +241,29 @@ target_strategise <- function(db) {
...
+target_emulator_config <- function(emulator_root) {
+  function() {
+    read_json(file.path(emulator_root, "config.json"))
+  }
+}
+
+endpoint_emulator_model <- function(emulator_root) {
+  print(emulator_root)
+  porcelain::porcelain_endpoint$new(
+    "GET", "/emulator/model/<filename>", target_emulator_model(emulator_root),
```
[the remark this answers is not in the document]
Comment (plietar) [reply R/api.R:260]: Yes the list of models is stored in the configuration. The configuration will probably look something like:
```yaml
{
    "models": [
        {
            "name": "FFNN", # User-friendly name, showed in the UI
            "filename": "FFNN.onnx", # What the frontend then passes to the `/model/<filename>`
            "preprocessing": { ... } # Metadata on how the form values should be converted to the vector of floats that is fed into the model
        },
        {
            "name": "LSTM (large)",
            "filename": "LSTM_large.onnx",
            "preprocessing": { ... }
        }
    ]
}
```

The `filename` effectively serves as the `emulatorType`, but avoids the need to maintain extra level of indirection inside of mintr.

[the remark this answers is not in the document]
Comment (EmmaLRussell) [reply R/api.R:259]: Fair enough - it's come from mintr in the first place, so the main thing is we don't need to hard-code it in mint.

**Comment by EmmaLRussell** [comment]
> Maybe we want to start including an emulator model (even a bad one) sooner than I'd expected, just so we have something to run frontend tests against. I'll think more about that.

Yeah, I think that's definitely going to be worth it for developing the front end where testing functionality doesn't require the emulator to be very accurate! It's either that or putting a script in mint to mount an emulator, but then we'd need to get the emulator from somewhere..

**Review by EmmaLRussell — APPROVED** [review]

**Pull request merged**